openedx/core/djangoapps/user_authn/cookies.py

```diff
@@ -38,34 +95,24 @@ def standard_cookie_settings(request):
         'httponly': None,
     }
 
+    # In production, TLS should be enabled so that this cookie is encrypted
+    # when we send it.  We also need to set "secure" to True so that the browser
+    # will transmit it only over secure connections.
+    #
+    # In non-production environments (acceptance tests, devstack, and sandboxes),
+    # we still want to set this cookie.  However, we do NOT want to set it to "secure"
+    # because the browser won't send it back to us.  This can cause an infinite redirect
+    # loop in the third-party auth flow, which calls `is_logged_in_cookie_set` to determine
+    # whether it needs to set the cookie or continue to the next pipeline stage.
+    cookie_settings['secure'] = request.is_secure()
+
     return cookie_settings
 
 
 def set_logged_in_cookies(request, response, user):
     """
-    Set cookies indicating that the user is logged in.
-
-    Some installations have an external marketing site configured
-    that displays a different UI when the user is logged in
-    (e.g. a link to the student dashboard instead of to the login page)
-
-    Currently, two cookies are set:
-
-    * EDXMKTG_LOGGED_IN_COOKIE_NAME: Set to 'true' if the user is logged in.
-    * EDXMKTG_USER_INFO_COOKIE_VERSION: JSON-encoded dictionary with user information (see below).
-
-    The user info cookie has the following format:
-    {
-        "version": 1,
-        "username": "test-user",
-        "header_urls": {
-            "account_settings": "https://example.com/account/settings",
-            "resume_block":
-                "https://example.com//courses/org.0/course_0/Run_0/jump_to/i4x://org.0/course_0/vertical/vertical_4"
-            "learner_profile": "https://example.com/u/test-user",
-            "logout": "https://example.com/logout"
-        }
-    }
+    Set cookies at the time of user login. See LOGGED_IN_COOKIE_NAMES to see
```

Comment:
Nit: PEP8 says the summary line should be a single line.  Doesn't always work out, but easy here to put the second sentence on its own line.